### PR TITLE
Top Level Metadata

### PR DESCRIPTION
**Pull request by kornypoet**
This PR adds `:Metadata` as a top-level attribute on `OrchestrationTemplate` as defined [here](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/metadata-section-structure.html).

It also moves `:Metadata` to an attribute (instead of a dsl object) on `Resource` to align behavior with the [documentation](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-attribute-metadata.html).

The above eliminates the need to have `Metadata` defined as a class; `metadata.rb` can be removed.

### CODE REVIEW

File: lib/cfndsl/resources.rb [hunk 1]
```diff
@@ -1,13 +1,12 @@
 require 'cfndsl/jsonable'
-require 'cfndsl/metadata'
 require 'cfndsl/properties'
 require 'cfndsl/update_policy'
 
 module CfnDsl
   # Handles Resource objects
   class ResourceDefinition < JSONable
-    dsl_attr_setter :Type, :DependsOn, :DeletionPolicy, :Condition
-    dsl_content_object :Property, :Metadata, :UpdatePolicy, :CreationPolicy
```
Comment (stevenjack) [inline lib/cfndsl/resources.rb]: @kornypoet Do we not want to keep this in for people that might still be using it and deprecate it before a major release? Or has it never been relevant as the metadata has always sat at the top level?

**Comment by kornypoet** [comment]
@stevenjack I am going to lean towards not, here's my reasoning:

With `:Metadata` a `dsl_content_object`, you have to pass in a handle and an argument. `to_json` on that resource creates a hash with only one top-level key. Metadata is allowed to be free-form JSON when supplied to CloudFormation. In its current implemented form, `:Metadata` is actually more restrictive than what the AWS documentation allows. This makes me view this change as more of a bug fix (although I can see how this might break existing templates) and is the correct way to move forward. To make a different argument, suppose a resource was incorrectly defined in the `yaml` types file. Correcting it might cause users' templates to "break", but it should be easy to correct and to understand the better behavior that AWS expects. I wouldn't deprecate behavior there either.

Honestly, I don't think anyone uses this feature at all. It was confusing to me to see the `:Metadata` as an object vs. an attribute, so I adjusted it to AWS's real-life expectations.

**Comment by kornypoet** [comment]
As soon as I posted that, I thought of a follow-up. I still don't think we need deprecation notices, but it does point out the holes in the coverage documentation for `cfndsl` and in the examples. Maybe including a sample usage there would alleviate most of the concern?

**Comment by kornypoet** [comment]
@stevenjack Happy to do this in the way you think is best.

**Comment by stevenjack** [comment]
@kornypoet agreed, could you add an example in the README and I'll get it merged

**Comment by kornypoet** [comment]
README updated @gergnz can you look this over?

File: lib/cfndsl/resources.rb (hunk 1 above)
Comment (gergnz) [inline lib/cfndsl/resources.rb]: I'm comfortable with removing this, I think @stevenjack your or statement makes sense, it's never really been relevant.

**Comment by gergnz** [comment]
@kornypoet makes sense.  Nothing stands out for me. Feel free to merge.

**Pull request merged**